Source/MQTTnet/Server/MqttClientConnection.cs

```diff
@@ -163,13 +167,30 @@ private async Task<MqttClientDisconnectType> RunInternalAsync()
                     Interlocked.Increment(ref _sentPacketsCount);
                     _lastPacketReceivedTimestamp = DateTime.UtcNow;
 
+                    if (packet is MqttAuthPacket clientAuthPacket)
+                    {
+                        if (_serverOptions.EnhancedAuthenticationBrokerHandler == null)
+                        {
+                            continue;
```

Comment:
I think this should force a disconnect rather than ignoring the message.

Reply:
Yes, I think this makes sense. It's a case which should never occur, though. How can I disconnect at this point?